addons/web/static/tests/helpers/test_env.js

```diff
@@ -15,17 +15,19 @@ odoo.define('web.test_env', async function (require) {
      * @returns {Proxy}
      */
     function makeTestEnvironment(env = {}, providedRPC = null) {
+
         const database = {
-            parameters: {
+            parameters: Object.assign({
                 code: "en_US",
                 date_format: '%m/%d/%Y',
                 decimal_point: ".",
                 direction: 'ltr',
                 grouping: [],
                 thousands_sep: ",",
                 time_format: '%H:%M:%S',
-            },
+            }, (env && env.database && env.database.parameters) ? env.database.parameters : {}),
```

Comment:
`env.database` is not clear, it should be something like "translation database" if you want it as a top level key, but that's probably not good either.

What we try to have here is that whatever we pass in `env` has the same structure as what it should replace, so `env._t.database`

Reply:
This doesn't look possible without trigger a `TypeError: this.env._t is not a function`.

Reply:
No test seems to pass a custom `_t`, so could be fine to always have `_t` being identity function (`s => s`), and you can customize `env._t_database.parameters` with
```js
env: {
   _t: {
      database: {
         parameters: {
            /* ... */
         },
      },
   },
},
```

Reply:
This make no sens to me.
`env._t` is a function or an object ?

Reply:
It's both. In JS all functions are objects that can contain props and methods.

Reply:
`_t` is a function (in JS, a function is an object, but an object is not necessarily a function). In the method `start()` you provide something to mainly set up `env`. You can provide an object, but the internal processing of param could do anything. Here I suggest to provide `_t` content all as object in `env`. The internal processing by `makeTestEnvironment()` will just turn `_t` as the identify function `s => s`, and will set other inner-provided params to this `_t` function (+ defaults for non-provided entries).

Reply:
I understand nothing.